src/postgraphile/postgraphile.ts

```diff
@@ -104,7 +115,10 @@ export function getPostgraphileSchemaBuilder<
     while (true) {
       try {
         if (options.watchPg) {
-          await watchPostGraphileSchema(pgPool, pgSchemas, options, newSchema => {
+          if (cancelWatchFn) {
+            throw new Error(`cancelWatchFn should not have been set at this point.`);
+          }
```

Comment:
It's not clear to me whether `watchPostGraphileSchema()` will ever be successfully invoked more than once. What I've got here is the brash thing, which is assert it doesn't happen. If we know it could be invoked more than once, then the thing to do, I think, would be invoke and nullify `cancelWatchFn` here instead.

Reply:
> I can't remember if `getPostgraphileSchemaBuilder` is a public API or not off-hand

It looks to me like it's not public. It's imported in `cli.ts` and `postgraphile.ts` and as far as I can tell it's not re-exported anywhere. Could you confirm? @benjie

Reply:
I checked this; it’s fine 👍